cmake/package_mac_productbuild.cmake

```diff
@@ -52,9 +52,23 @@ if(NOT DEFINED CPACK_PKGBUILD_IDENTITY_NAME)
   message(WARNING "CPACK_PKGBUILD_IDENTITY_NAME not set. PKG will not be signed. Make sure to specify an identity with a Developer ID: Installer certificate (not Application certificate).")
 endif()
 
+set(MACOS_TARGET_ARCHS ${CMAKE_OSX_ARCHITECTURES})
+if (NOT MACOS_TARGET_ARCHS)
+  # Warning: if cmake is a subprocess of a process that is run under Rosetta, it will yield
+  #  x86_64 (but probably also build for it. Therefore it should be fine.)
+  set(MACOS_TARGET_ARCHS ${CMAKE_HOST_SYSTEM_PROCESSOR})
+endif()
+if (MACOS_TARGET_ARCHS STREQUAL "x86_64")
+  set(ARCH_SUFFIX "Intel")
+elseif (MACOS_TARGET_ARCHS STREQUAL "arm64")
+  set(ARCH_SUFFIX "Silicon")
+elseif ("x86_64" IN_LIST MACOS_TARGET_ARCHS AND "arm64" IN_LIST MACOS_TARGET_ARCHS)
+  set(ARCH_SUFFIX "Universal")
+else ()
+  set(ARCH_SUFFIX "Unknown")
```

Comment:
Same as my comment on the dragndrop cmake file. I think we should notify the user that something is funky.